packages/nimble-components/src/table-column/enum-base/models/mapping-icon-config.ts

```diff
@@ -13,19 +13,20 @@ export interface IconView {
 const emptyTemplate = html<IconView>` `;
 
 const createIconTemplate = (
-    icon: string | undefined
+    icon: string | undefined,
+    textHidden: boolean
 ): ViewTemplate<IconView> => {
     if (icon === undefined) {
         return emptyTemplate;
     }
 
     return html`
         <${icon}
-            title="${x => x.text}"
+            title="${x => (textHidden ? x.text : '')}"
```

Comment:
same as the spinner we should move towards fewer unique instances instead of further. If we follow a similar patterns as the spinner and make x.textHidden part of the config we at least remove one dimension of uniqueTemplate construction. Longer-term we may want to cache the template instances per icon (but so far has seemed like an unneeded optimization).